### DISCUSSION TITLE
mcmc obtained two locally converged solutions

### DISCUSSION THREAD

**Discussion by 0batsy0** [phoebe-project/phoebe2#798] [Questions]
After running 2000 iterations, I found that it converged to two positions. I am not sure how to deal with such a problem next, or whether I can just ignore it.
![1](https://github.com/phoebe-project/phoebe2/assets/130527283/ca93a650-cb88-4421-9d1f-38e0a669cbf1)
![捕获](https://github.com/phoebe-project/phoebe2/assets/130527283/6c8cced2-aa37-4f7a-83b0-fc8b915557ab)

**Comment by aprsa** [comment]
Yep, this can happen: some of the walkers are stuck in a local minimum (bottom branch). This is demonstrably a poorer solution of the two (lower logp). The walkers should eventually flop to the better minimum but it can take a loooooong time. The more practical option is to either use a logp cutoff when exploring your solution so that you only consider walkers in the upper branch, or resample all walkers from the upper branch and continue iterations from there. In general the second option I believe is better, except when only up to a few walkers got stuck in a local minimum.

**Reply by 0batsy0** [reply]
Thank you for your reply. Since our data only has a very small number of nwalkers, consider the first processing solution you mentioned. Can it be understood as choosing to abandon nwalkers that fall into local optima? In the case where the autocorrelation time shows convergence, can the results obtained in this way be used in the final results? Attached is the result of this processing:
![捕获](https://github.com/phoebe-project/phoebe2/assets/130527283/a9991918-004f-45ac-8610-b3c96a32c5de)

![捕获](https://github.com/phoebe-project/phoebe2/assets/130527283/8bbc2094-5f30-411b-8d56-2fcef7551b9e)

**Comment by aprsa** [comment]
Yes, they can be, provided that the number of "lost" walkers is small compared to the overall number of sampled parameters (which seems to be the case). I would consider these posteriors reasonable, but you would need to assess convergence before considering them final.

**Reply by 0batsy0** [reply]
Thank you very much for your answer, it helped me have a deeper understanding of mcmc.
